frontend/main.js

```diff
@@ -101,11 +101,30 @@ async function startApp () {
       L('Fatal error: {reportError}', LError(e)), 'exclamation-triangle'
     )
   }
-  sbp('chelonia/configure', {
+  await sbp('chelonia/configure', {
     connectionURL: sbp('okTurtles.data/get', 'API_URL'),
     stateSelector: 'state/vuex/state',
     reactiveSet: Vue.set,
     reactiveDel: Vue.delete,
+    contracts: {
+      defaults: {
+        modules: { '@common/common.js': Common },
+        allowedSelectors: [
+          'state/vuex/state', 'state/vuex/commit', 'state/vuex/getters',
+          'chelonia/contract/sync', 'chelonia/contract/remove', 'controller/router',
+          'gi.actions/identity/updateLoginStateUponLogin',
+          'gi.actions/chatroom/leave', 'gi.notifications/emit'
+        ],
+        allowedDomains: ['okTurtles.data', 'okTurtles.events', 'okTurtles.eventQueue'],
+        preferSlim: true
+      },
+      manifests: {
+        'gi.contracts/group': '21XWnNSHSo5Dov3RwKFwSJzMNpt3KmpJNaVN6qsacysUnmxgq9',
+        'gi.contracts/identity': '21XWnNVUKkh1dWq1RoAAJbkPnGXxqFUpjXrzb9yo7NeGEEToin',
+        'gi.contracts/mailbox': '21XWnNKoSm7e2vnGJvoFSRB7heFTeg24FwDCkb6JAazkzhpNdY',
+        'gi.contracts/chatroom': '21XWnNXJkm2bkc8Wu7TC38zB845tR28qRGCs2HXCeMEJgvFozi'
```

Comment:
In my opinion, `grunt pin` should generate a manifest JSON file in a specific format and main.js should read it.
Instead of updating main.js.

Reply:
That's an excellent suggestion @Silver-IT! Not sure why I didn't think to do that!